test/parallel/test-child-process-uid-gid.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const spawn = require('child_process').spawn;
+
+assert.throws(() => {
+  spawn('echo', ['fhqwhgads'], {uid: 0});
+}, /EPERM/, 'Setting UID should throw EPERM for unprivileged users.');
+
+assert.throws(() => {
+  spawn('echo', ['fhqwhgads'], {gid: 0});
+}, /EPERM/, 'Setting GID should throw EPERM for unprivileged users.');
```

Comment:
> I'm reasonably sure these are going to fail with ENOTSUP on Windows.

Sure enough: https://ci.nodejs.org/job/node-test-binary-windows/2405/RUN_SUBSET=2,VS_VERSION=vcbt2015,label=win10/tapTestReport/test.tap-23/

Will add a check for `common.isWindows` and alter the expected error appropriately.